File: Backend/Backend.WebServices/DatabaseEntities/Place.cs
namespace Backend.WebServices.DatabaseEntities
{
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;

    [Table("Place")]
    public partial class Place
    {
        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2214:DoNotCallOverridableMethodsInConstructors")]
        public Place()
        {
            MediaItems = new HashSet<MediaItem>();
        }

        public int Id { get; set; }

        public int ParticipantId { get; set; }

        public int SequenceNumber { get; set; }

        [StringLength(255)]
        public string Name { get; set; }

        public decimal? Latitude { get; set; }

        public decimal? Longitude { get; set; }

        public string Description { get; set; }

        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<MediaItem> MediaItems { get; set; }

        public virtual Participant Participant { get; set; }
    }
}

File: Frontend/app/shared/place.ts
import {MediaItem} from "./mediaItem";
import {SimpleMediaItem} from "./simpleMediaItem";

export class Place
{
    public mediaItems : Array<MediaItem> = [];

    constructor(
        public id: number, 
        public sequenceNumber: number, 
        public name: string, 
        public latitude: number, 
        public longitude: number, 
        public description: string)
    {

    }

    getMediaToDownload() : Array<SimpleMediaItem>
    {
        var result = [];
        for (let mediaItem of this.mediaItems){
            result.push(new SimpleMediaItem(mediaItem.filename, mediaItem.url));
        }
        return result;
    }
}